overlays/tools.nix

```diff
@@ -121,5 +121,26 @@ in { haskell-nix = prev.haskell-nix // {
             })
           ];
         })).ghcide.components.exes.ghcide;
+
+    haskell-language-server."0.5.1" = args:
+      (final.haskell-nix.cabalProject ( args // {
+        name = "haskell-language-server";
+        src = final.fetchFromGitHub {
+          owner = "haskell";
+          repo = "haskell-language-server";
+          rev = "0.5.1";
+          sha256 = "17nzgpiacmrvwsy2fjx6a6pcpkncqcwfhaijvajm16jpdgni8mik";
+          fetchSubmodules = true;
+        };
+        sha256map = {
+          "https://github.com/bubba/brittany.git"."c59655f10d5ad295c2481537fc8abf0a297d9d1c" = "1rkk09f8750qykrmkqfqbh44dbx1p8aq1caznxxlw8zqfvx39cxl";
+        };
+        # Plan issues with the benchmarks, can try removing later
+        configureArgs = "--disable-benchmarks";
+        modules = [{
+          # Tests don't pass for some reason, but this is a somewhat random revision.
+          packages.haskell-language-server.doCheck = false;
+        }];
```

Comment:
When I built it myself, I found that these lines weren't necessary.

Reply:
This bit was copied from the Plutus repo and I think the `configureArgs = "--disable-benchmarks"` makes it work with ghc 8.10.  When I fixing up the hls build test in haskell.nix for GHC 8.10 I got it working with https://github.com/input-output-hk/haskell.nix/blob/97b8e28ca67af73b69bd7115dd17f2f84e29e0da/test/haskell-language-server/cabal.nix#L15-L20
Disabling the benchmarks seems simpler though.

I suspect `packages.haskell-language-server.doCheck = false` might be unnecesary (perhaps plutus was explicitly running the tests on all its tools).

Reply:
Yeah, it's the same workaround as your cabalProjectLocal thing, but I thought this way was simpler. I don't remember why we needed `doCheck = false`. Poor commenting by past me!